### ISSUE TITLE
Add Twinax cable support

### ISSUE THREAD

**Issue by swerveshot** [digitalocean/netbox#2682]
### Change Type
[X] Addition
[ ] Correction
[ ] Deprecation
[ ] Cleanup (formatting, typos, etc.)

### Environment
Python version: 3.6.7
NetBox version: v2.5.0

### Proposed Functionality
When observing the available cable types in the connection section of the devices menu I noticed the usual suspects for copper and optical fibre. But I'm missing a probably less used one. A Cisco SFP+ Twinax copper cable. Details can be found here: https://www.cisco.com/c/en/us/products/collateral/interfaces-modules/transceiver-modules/data_sheet_c78-455693.html

Could you please add this cable type to netbox please? I'll try to to it myself but haven't looked at the files related to this type of change before.

**Comment by DanSheps** [comment]
Here is all the code that would probably be required:

In netbox\dcim\constants.py:

`CABLE_TYPE_TWINAX=1100
CABLE_TYPE_DAC=1200
CABLE_TYPE_AOC=3900`

add to the start of the array for "copper":

`(CABLE_TYPE_TWINAX,'Twinax Cable'),
(CABLE_TYPE_DAC,'SFP/SFP+ Direct Attached Cable'),`

add to the start of the array for "fiber":

`(CABLE_TYPE_AOC,'SFP/SFP+ Active Optical Cable'),`

Of course, @jeremystretch might suggest better #'s for the ID's

**Comment by mladewig** [comment]
Just came across this today. Would love to have aoc and dac as cable options.

**Comment by jeremystretch** [comment]
I don't think we would have separate types for DAC and Twinax, would we? My (possibly wrong) understanding is that Twinax is the physical medium employed by direct-attach copper cabling (DAC).

Seems like there should be three classes of direct-attach cabling:

* Passive copper
* Active copper
* Active optical (AOC)

Hopefully someone with more hands-on experience can chime in.

**Comment by rdegez** [comment]
As DAC cables kind of embed transceivers on each side I guess we need various declinaison (for each types of supported transceiver interface).

Most common would be 
- Active/passive coper & optical SFP/SFP+DAC (1G/10G)
- Active/passive coper & optical QSFP DAC (40G)
- Active/passive coper & optical QSFP28 DAC (100G)

To complicate things further : a lot of DAC cables types are actually not symetrical.

Some switchs support "agileports" with the ability to split à single QSFP port to 4x10Gbe link for instance.
QSFP breakout to 4 x SFP+ DACs are very common as it's a VERY cheap way to connect 4 servers with 10Gbe NIC to a single QSFP switch port. QSFP28 breakout to 4xSFP28 as well.

Supporting them would require 1:N mapping. Don't know if the netbox datamodel can handle these.

And what about fiber breakout cables with say 12/24/48 MMF or SMF in a single cable ? They would require N:N mapping :-)

**Comment by jeremystretch** [comment]
@rdegez The only thing we're concerned with here is the medium classification; interface speed is represented by the interface type. Additionally, NetBox does not model breakout cables; each channel in a 4x10GE breakout should be modeled as a separate interface with its own cable.

**Comment by rdegez** [comment]
@jeremystretch

Ok, thanks for the clarification on breakout cables. I guess this limitation can be overcome by specifing the same cable label to group these. (and creating 'subinterfaces' directly in the device as you said)

But the need for differentiation between SFP/SFP+, QSFP & QSFP28 DAC cables stands I guess ?
Otherwise why differentiate cat4/5/5e/6 RJ45 copper cables ?

**Comment by mladewig** [comment]
I agree with direct attach top level cable type with passive copper, active copper, and active optical as selectable options. Twinax is copper cable type used in either active or passive copper DACs. It could be said these are active or passive twinax cables. That said AOC is not refered to as active multimode fiber. I would suggest going with the more generic copper/optical terms.

**Comment by DanSheps** [comment]
Yeah, I don't really use active copper as generally at the lengths where I need active I am using optical.  I updated my code, but if no one else wants to implement this I can do the PR since it is fairly straight forward.

**Comment by zsandrus** [comment]
We use these cables fairly heavily in our environment. This includes Active and Passive DAC and AOC cables for SFP/SFP+/SFP28/QSFP/QSFP28 connections. I don't see any reason to separate the connector type since that is on the interface, but I think that the three cable types should be available.
